### ISSUE TITLE
Improve error for non-program exercises

### ISSUE THREAD

**Issue by faloi** [uqbar-project/mumuki-gobstones-server#36]
Right now, this is what you get when you submit a `program` in a procedure/function exercise:

```
Cannot define program "program"
program "program" already defined en:
/tmp/gobstones.20150610-3-1f6ek54.gbs (línea 1, columna 1)
```

It would be nice to add a default expectation for every non-program exercise that checks that `program` is not defined.

**Comment by faloi** [comment]
@flbulgarelli, please comment on this

**Comment by flbulgarelli** [comment]
Yup, agree. It could be a `HasProgram`, and then negate it `Not:HasProgram`.

Now that we are working on feedback, we could augment this output adding a feedback on duplicated elements. We could have the following rules:

 * If elements duplicated are in code and extra: "Please notice we are already providing the program. You just need to define procedure Zaraza. Remove that program declaration"
* If elements are duplicated within code: "Please notice that you have defined Zaraza twice. Try by having only one declaration of it".

**Comment by flbulgarelli** [comment]
Another option, instead of having `HasProgram`, `HasFunction`, `HasProcedure` is using `HasBinding:Target`, which is consistent with other runners
